### DISCUSSION TITLE
Sympy Sin wont work

### DISCUSSION THREAD

**Discussion by Sleppzz** [sympy/sympy#26610] [General]
I tryed sympy for a project, but it wont work. If i try to calculate sin(90) it outputs 0

Heres a little code i created, to show this:
`import sympy`
`a = (int(sympy.sin(90)))`
`b = (int(sympy.cos(90)))`
`print(a, "+", b, "=", a + b)`

The output is:
`0 + 0 = 0`

**Comment by moorepants** [comment]
You may mean to provide this input:

```
In [1]: import sympy

In [2]: a = sympy.sin(sympy.pi/2)

In [3]: b = sympy.cos(sympy.pi/2)

In [4]:  print(a, "+", b, "=", a + b)
1 + 0 = 1
```

**Comment by moorepants** [comment]
`sin()` and `cos()` take inputs in the unit of radian and if you use `int()` you may round values inadvertently.

**Reply by kompre** [reply]
> `sin()` and `cos()` take inputs in the unit of radian

I have a bone to pick with this statement though*, because the trigonometric functions will take ***implied*** radians units. I mean if we use the `convert_to(expr, radians)` function from the physics module to convert degree to radians, we will get an object like `pi/2 * rad` which even if is technically correct, will not be correctly evaluated by the trigs functions.

The correct way to convert degree to radians for use in the trigs function is to `convert_to(expr, 1)` which is a bit counter intuitive in my opinion, and can create issue down the line:

- let's say I wrote a symbolic expression like `expr = sin(alpha)` 
- I define a parameter dictionary where `param = {alpha: 30*degree}` with `degree` being the unit from the physics module.
- then when I substitute the dict into the expression `expr.subs(param)`  I will get `sin(30*degree)` back, which cannot be evaluated.

I found out that it is better to use `convert_to(x, 1)` before the substitution because with fairly complex expression can fail to evaluate correctly. This means I need to insert a new step where I convert degree to radians (I prefer doing this as soon as I defined the parameters dict where degrees appears)

It would be nice if the trigs function will evaluate directly also expressions that contains degree and radians unit, and not rely only on the implicit radian unit. 

> \* not with you of course
